### PR TITLE
Prepare v0.50.0 release

### PR DESCRIPTION
**Pull request by ashutosh-narkar**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,27 +3,118 @@
...
-- The chain of metadata returned by the `rego.metadata.chain()` built-in function will now contain an entry for the package even if the annotations are declared in another file, if the scope is `package`. 
+### Runtime, Tooling, SDK
+
+- server: Support compression of response payloads if HTTP client supports it (#5696) authored by @AdrianArnautu
+- bundle: Ensure the bundle resulting from merging a set of bundles does not contain `nil` data (#5709)
+- repl: Use lowercase for repl commands only and keep any provided arguments as-is (#5698) authored by @Trolloldem
+- internal/prometheus: New endpoint `/metrics/alloc_bytes` to show OPA's memory utilization (#5715)
+- server: When using OPA TLS authorization, authz policy authors will now have access to the client certificates
+presented as part of the TLS connection. This new data will be available under the key `client_certificates` (#5742)
+- server: Use streaming implementation of json.Decode rather than unmarshalling the incoming request (#5661)
```
Comment (anderseknert) [inline CHANGELOG.md:78]: "rather than using an intermediate buffer for the incoming request"

**Review by anderseknert — COMMENTED** [review]
Some comments but looks good. This will be a great release!

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,27 +3,118 @@
...
 project adheres to [Semantic Versioning](http://semver.org/).
 
-## Unreleased
+## 0.50.0
 
-### Runtime
+This release contains a mix of new features, bugfixes, security fixes, optimizations and build updates related to
+OPA's published images.
+
+### New Built-in Function: JSON Schema Verification and Validation
```
[the remark this answers is not in the document]
Comment (ashutosh-narkar) [reply CHANGELOG.md]: Done

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,27 +3,118 @@
...
 either with the `--user` argument for `docker run`, or by specifying the `securityContext` in the Kubernetes Pod specification.
 
-The change to the `package` scope in [#5251](https://github.com/open-policy-agent/opa/issues/5251) may cause compile-time errors and behavioural changes to type checking when the `schemas` annotation is used, and to rules calling the `rego.metadata.chain()` built-in function:
 
-- Existing projects with the same package declared in multiple files will trigger a `rego_type_error: package annotation redeclared` error _if_ two or more of these are annotated with the `package` scope.
-- If using the `package` scope, the `schemas` annotation will be applied to type checking also for rules declared in another file than the annotation declaration, as long as the package is the same.
-- The chain of metadata returned by the `rego.metadata.chain()` built-in function will now contain an entry for the package even if the annotations are declared in another file, if the scope is `package`. 
+### Runtime, Tooling, SDK
+
+- server: Support compression of response payloads if HTTP client supports it (#5696) authored by @AdrianArnautu
```
[the remark this answers is not in the document]
Comment (ashutosh-narkar) [reply CHANGELOG.md]: Updated the PR/Issue links

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,27 +3,118 @@
...
+- bundle: Ensure the bundle resulting from merging a set of bundles does not contain `nil` data (#5709)
+- repl: Use lowercase for repl commands only and keep any provided arguments as-is (#5698) authored by @Trolloldem
+- internal/prometheus: New endpoint `/metrics/alloc_bytes` to show OPA's memory utilization (#5715)
```
[the remark this answers is not in the document]
Comment (ashutosh-narkar) [reply CHANGELOG.md]: Done

File: CHANGELOG.md (hunk 1 above)
Comment (ashutosh-narkar) [reply CHANGELOG.md]: Done

**Review by anderseknert — APPROVED** [review]

**Pull request merged**